log/logtest/factory_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logtest
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/exp/slices"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+func TestRecordFactory(t *testing.T) {
+	now := time.Now()
+	observed := now.Add(time.Second)
+	severity := log.SeverityDebug
+	severityText := "DBG"
+	body := log.StringValue("Message")
+	attrs := []log.KeyValue{
+		log.Int("int", 1),
+		log.String("str", "foo"),
+		log.Float64("flt", 3.14),
+	}
+
+	got := RecordFactory{
+		Timestamp:         now,
+		ObservedTimestamp: observed,
+		Severity:          severity,
+		SeverityText:      severityText,
+		Body:              body,
+		Attributes:        attrs,
+	}.NewRecord()
+
+	assert.Equal(t, now, got.Timestamp())
+	assert.Equal(t, observed, got.ObservedTimestamp())
+	assert.Equal(t, severity, got.Severity())
+	assert.Equal(t, severityText, got.SeverityText())
+	assert.Equal(t, body, got.Body())
```

Comment:
It is safer to assert body using `Equal`. See: https://github.com/open-telemetry/opentelemetry-go/blob/baeb560673c8ff82789dbdeaaad525418c70337f/log/keyvalue.go#L41-L42

AFAIK testify uses reflect to compare. This assertion may start failing if we change the internals.

I propose to copy the `assertBody` from here:
https://github.com/open-telemetry/opentelemetry-go/blob/59f0956d18e9392b358f6117b6c8f05ee3f45e48/sdk/log/logtest/factory_test.go#L106-L112